artifacts/deployment/02-sample-apiserver-certs.yaml

```diff
@@ -0,0 +1,104 @@
+# This manifest deploys certificates for the sample-apiserver and the proxied etcd,
+# and RBAC roles for the EtcdProxy Controller to manage the certificates in the API server namespace.
+#
+# The CA certificate for the sample-apiserver is stored in the 'sample-apiserver-ca' ConfigMap.
+# The client certificate and key for the sample-apiserver, which can be used by users to access the API,
+# are stored in the 'sample-apiserver-clientcert' Secret.
+#
+# The CA certificate for the proxied etcd is stored in the 'etcd-serving-ca' ConfigMap,
+# and the client certificate and key are stored in the 'etcd-client-cert' Secret.
+#
+# The etcd certificates are managed by the EtcdProxy Controller, so this manifests
+# creates RBAC roles allowing the EtcdProxy Controller ServiceAccount to get, update and patch
+# ConfigMap and Secret containing the etcd certificates.
+---
+# The CA certificate for the sample-apiserver. Allows users to use client certificates to access the API.
```

Comment:
we can't read the one from kube-public or kube-system?  You can have it for now, but it's weird.

Reply:
Yes, we can, and it's the default behavior. What I wanted is to enable users to use API outside, but, I can agree it's bit weird.

Do you want me to get rid of those?

Reply:
> Yes, we can, and it's the default behavior. What I wanted is to enable users to use API outside, but, I can agree it's bit weird.

Assuming it wires well by default, you can use your kube client cert to speak directly to the sample apiserver.

Reply:
Removed them for now. It should be the default behavior